sourcetool/pkg/auth/authenticator.go

```diff
@@ -0,0 +1,153 @@
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+
+package auth
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/google/go-github/v69/github"
+)
+
+const (
+	// GitHub endpoints
+	deviceCodeURL = "https://github.com/login/device/code"
+	tokenURL      = "https://github.com/login/oauth/access_token" //nolint:gosec // not a credential
+
+	// The SLSA sourcetool app OAuth client ID
+	oauthClientID = "Ov23lidVQsiU5R5tod3z"
+
+	// App's config directory name
+	configDirName = "slsa"
```

Comment:
nit: should this not be a hidden dir? Maybe something under `XDG_DATA_HOME` or `XDG_CONFIG_HOME`? consider also renaming to `dataDirName` this since we seem to be saving data like the oauth token to this directory.

Reply:
Oh, ok. just read that as UserHomeDir. I would be happier if it was under XDG_DATA_HOME since this is state data but I think this is ok for now.